src/complex/Filter/Actions/MoveDataAction.cpp

```diff
@@ -54,6 +60,11 @@ Result<> MoveDataAction::apply(DataStructure& dataStructure, Mode mode) const
     return MakeErrorResult(k_MissingParentObjectCode, fmt::format("{}Could not find a possible parent object at '{}'", prefix, m_NewParentPath.toString()));
   }
 
+  if(const auto* dataGroup = dataStructure.getDataAs<BaseGroup>(m_Path); dataGroup != nullptr && dataGroup->isParentOf(newParent))
```

Comment:
Well I checked trying to move a grandparent into one of its grandchildren and this check caught that. I think maybe because the DataObject::hasParent function is using that StringUtilities::contains function. I could write a slightly different recursive approach if we think that'll be more robust though.